Add custom SQA entries specific to the electromagnetic module and improve defaults/initialization

cticenhour opened the pull request:
This PR also makes adjustments to framework SQA docs and the templates for formatting and defaults as well as adding a new module option to the MooseDocs SQA initialization command.

Refs #15968
Refs #20693

cticenhour committed:
Make adjustments to framework sqa docs and app sqa template

Refs #15968
Refs #20693

cticenhour committed:
Fill-in missing sqa document fields customized for the EM module

Refs #15968
Refs #20693

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1198006/) on fe1e6e1 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/22372/site)

This comment will be updated on new commits.

cticenhour committed:
Add modules sqa document templates

Refs #15968
Refs #20693

cticenhour committed:
Update em module sqa docs to improve language and module srs template to add more sections that could work for all modules

Refs #15968
Refs #20693

cticenhour committed:
Fixup MooseDocs SQA initialization to better handle MOOSE modules versus apps

`./moosedocs.py init sqa --module MODULE` now produces a specific set of initial files for modules
with the `--app APP` argument reproducing prior behavior

Refs #15968

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1200691/) on 9b16502 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>c62e4c</th>
      <th colspan=4>#22372 9b1650</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.30%</td>
      <td>84.30%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>80164</td>
      <td>80162</td>
      <td>-2</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>14927</td>
      <td>14929</td>
      <td>+2</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/22372/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/22372/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/richards)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/22372/coverage/xfem)

</details>



This comment will be updated on new commits.

cticenhour committed:
Update module config.yml template and improve initial module setup message for improved SQA procedures

permcody reviewed (approved):
This looks good! Thanks!
